src/sparkle/application/__init__.py

```diff
@@ -1,68 +1,132 @@
+            config (Config): Configuration object containing application-specific settings.
+            writers (List[Writer]): List of Writer objects used for writing the processed data.
         """
-        self.spark_session = spark_session
+        self.env = env
+        self.spark_config = spark_config
         self.config = config
         self.writers = writers
+        self.spark_session = self.get_spark_session(env)
+
+    def get_spark_session(self, env: ExecutionEnvironment) -> SparkSession:
+        """Create and return a Spark session based on the environment.
+
+        Args:
+            env (ExecutionEnvironment): The environment type (either LOCAL or AWS).
+
+        Returns:
+            SparkSession: The Spark session configured with the appropriate settings.
+        """
+        if env == ExecutionEnvironment.LOCAL:
+            return self._get_local_session()
+        elif env == ExecutionEnvironment.AWS:
+            return self._get_aws_session()
+        else:
+            raise ValueError(f"Unsupported environment: {env}")
+
+    def _get_local_session(self) -> SparkSession:
+        """Create a Spark session for the local environment."""
+        ivy_settings_path = os.environ.get("IVY_SETTINGS_PATH", None)
+        LOCAL_CONFIG = {
+            "spark.sql.extensions": ",".join(_SPARK_EXTENSIONS),
+            "spark.jars.packages": ",".join(_SPARK_PACKAGES),
+            "spark.sql.session.timeZone": "UTC",
+            "spark.sql.catalog.local": "org.apache.iceberg.spark.SparkCatalog",
+            "spark.sql.catalog.local.type": "hadoop",
+            "spark.sql.catalog.local.warehouse": "./tmp/warehouse",
+        }
+
+        spark_conf = SparkConf()
+        for key, value in LOCAL_CONFIG.items():
+            spark_conf.set(key, str(value))
+        for key, value in self.spark_config.items():
+            spark_conf.set(key, str(value))
+
+        spark_session_builder = SparkSession.builder.master("local[*]").appName(
+            "LocalDataProductApp"
+        )
+        for key, value in spark_conf.getAll():
+            spark_session_builder = spark_session_builder.config(key, value)
+
+        return spark_session_builder.getOrCreate()
+
+    def _get_aws_session(self) -> SparkSession:
+        """Create a Spark session for the AWS environment."""
+        try:
+            from awsglue.context import GlueContext
+        except ImportError:
+            logger.error("Could not import GlueContext. Is this running on AWS Glue?")
+
+        AWS_CONFIG = {
+            "spark.sql.extensions": ",".join(_SPARK_EXTENSIONS),
+            "spark.sql.catalog.glue_catalog": "org.apache.iceberg.spark.SparkCatalog",
+            "spark.sql.catalog.glue_catalog.catalog-impl": "org.apache.iceberg.aws.glue.GlueCatalog",
+            "spark.sql.catalog.glue_catalog.io-impl": "org.apache.iceberg.aws.s3.S3FileIO",
+            "spark.sql.catalog.glue_catalog.warehouse": "./tmp/warehouse",
```

Comment:
@farbodahm this is agains the whole idea of sparkle remains cloud neutral. This configuration must be passed to the sparkle so the sparkle can get or create the spark session with knowing if its coming from glue or synapse or a local system.

Reply:
Yeah, I agree with @kkiani, however:
1. moving it is not that expensive, all of it is text config, and Sparkle can expect it as a set of extra items.
2. Sparkle being cloud-agnostic, is different from having no knowledge about the cloud. Sparkle can import related libraries or have some setup, if special configuration related to a cloud is passed to it. If not, works with general configuration.